lib/src/ReconnectionManager.dart

```diff
@@ -27,26 +27,42 @@ class ReconnectionManager {
       handleReconnection();
     } else if (state == XmppConnectionState.SocketOpening) {
       //do nothing
-    } else if (state != XmppConnectionState.Reconnecting) {
-      isActive = false;
-      timeOutInMs = initialTimeout;
-      counter = 0;
-      if (timer != null) {
-        timer!.cancel();
-        timer = null;
-      }
+    } else if ([
```

Comment:
@slashdigital i fixed a bit on reconnection , not very ideal, but it does some jobs.